templates/terraform/resource.erb

```diff
@@ -217,21 +217,23 @@ func resource<%= resource_name -%>Create(d *schema.ResourceData, meta interface{
 
 <%  if !object.async.nil? && object.async.allow?('create') -%>
 <%    if object.async.is_a? Api::PollAsync -%>
-    waitURL, err := replaceVars(d, config, "<%= "{{#{object.__product.name}BasePath}}#{object.self_link_uri}" -%>")
+    pollURL, err := replaceVars(d, config, "<%= "{{#{object.__product.name}BasePath}}#{object.self_link_uri}" -%>")
     if err != nil {
         return err
     }
 
-    err = <%= client_name_camel -%>PollingWaitTime(
-    config, res,<% if has_project -%> project, <% end -%> waitURL, "Creating <%= object.name -%>",
-        int(d.Timeout(schema.TimeoutCreate).Minutes()))
-
+    err = PollingWaitTime(config, <%= client_name_camel -%>PollWaiter(res), <% if has_project -%> project, <% end -%> pollURL, "Creating <%= object.name -%>", d.Timeout(schema.TimeoutCreate))
```

Comment:
There isn't a `PollingWaitTime` that won't accept project, so if has_project is false this will generate broken code.

Reply:
good catch!